temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowThreadContext.java

```diff
@@ -123,33 +130,35 @@ private void maybeEvaluateLocked(String reason) {
       } catch (Exception e) {
         evaluationFunction.apply(Throwables.getStackTraceAsString(e));
       } finally {
-        status = Status.YIELDED;
+        status = Status.RUNNING;
```

Comment:
It's a more correct state to switch after evaluation. If running conditions are not met, we will go into YIELDED anyway after the check.

Before this change the following chains of states were possible: 
EVALUATING -> YIELDED
EVALUATING -> YIELDED (shortly) -> RUNNING and YIELDED will be set and observable without actually yielding. Ideally, we want to switch to and publish YIELDED state only when we ACTUALLY yield.

After the change, we can get 
EVALUATING -> RUNNING (shortly) -> YIELDED 
or EVALUATING -> RUNNING
and we go into YIELDED ONLY if we actually yield. This is a much more consistent state model.